packages/connection/src/class-manager.php

```diff
@@ -1382,18 +1415,96 @@ public function handle_authorization() {
 
 	/**
 	 * Builds a URL to the Jetpack connection auth page.
-	 * This needs rethinking.
-	 *
-	 * @param bool        $raw If true, URL will not be escaped.
-	 * @param bool|string $redirect If true, will redirect back to Jetpack wp-admin landing page after connection.
-	 *                              If string, will be a custom redirect.
-	 * @param bool|string $from If not false, adds 'from=$from' param to the connect URL.
-	 * @param bool        $register If true, will generate a register URL regardless of the existing token, since 4.9.0.
 	 *
-	 * @return string Connect URL
+	 * @param WP_User $user (optional) defaults to the current logged in user.
+	 * @param String  $redirect (optional) a redirect URL to use instead of the default.
+	 * @return string Connect URL.
 	 */
-	public function build_connect_url( $raw, $redirect, $from, $register ) {
-		return array( $raw, $redirect, $from, $register );
+	public function get_authorization_url( $user = null, $redirect = null ) {
+
+		if ( empty( $user ) ) {
+			$user = wp_get_current_user();
+		}
+
+		$roles       = new Roles();
+		$role        = $roles->translate_user_to_role( $user );
+		$signed_role = $this->sign_role( $role );
+
+		/**
+		 * Filter the URL of the first time the user gets redirected back to your site for connection
+		 * data processing.
+		 *
+		 * @since 8.0.0
+		 *
+		 * @param string $redirect_url Defaults to the site admin URL.
+		 */
+		$processing_url = apply_filters( 'jetpack_connect_processing_url', admin_url( 'admin.php' ) );
+
+		/**
+		 * Filter the URL to redirect the user back to when the authorization process
+		 * is complete.
+		 *
+		 * @since 8.0.0
+		 *
+		 * @param string $redirect_url Defaults to the site URL.
+		 */
+		$redirect = apply_filters( 'jetpack_connect_redirect_url', $redirect );
+
+		$secrets = $this->generate_secrets( 'authorize', $user->ID, 2 * HOUR_IN_SECONDS );
+
+		/**
+		 * Filter the type of authorization.
+		 * 'calypso' completes authorization on wordpress.com/jetpack/connect
+		 * while 'jetpack' ( or any other value ) completes the authorization at jetpack.wordpress.com.
+		 *
+		 * @since 4.3.3
+		 *
+		 * @param string $auth_type Defaults to 'calypso', can also be 'jetpack'.
+		 */
+		$auth_type = apply_filters( 'jetpack_auth_type', 'calypso' );
+
+		/**
+		 * Filters the user connection request data for additional property addition.
+		 *
+		 * @since 8.0.0
+		 *
+		 * @param Array $request_data request data.
+		 */
+		$body = apply_filters(
+			'jetpack_connect_request_body',
+			array(
+				'response_type' => 'code',
+				'client_id'     => \Jetpack_Options::get_option( 'id' ),
+				'redirect_uri'  => add_query_arg(
+					array(
+						'action'   => 'authorize',
+						'_wpnonce' => wp_create_nonce( "jetpack-authorize_{$role}_{$redirect}" ),
+						'redirect' => rawurlencode( $redirect ),
```

Comment:
I suspect we're changing to rawurlencode based on phpcs and WP Coding [standards](https://github.com/WordPress/WordPress-Coding-Standards/blob/master/WordPress/Sniffs/PHP/DiscouragedPHPFunctionsSniff.php#L49)